src/jit/codegencommon.cpp

```diff
@@ -10204,7 +10204,19 @@ var_types Compiler::GetHfaType(CORINFO_CLASS_HANDLE hClass)
     {
 #ifdef FEATURE_HFA
         CorInfoType corType = info.compCompHnd->getHFAType(hClass);
-        if (corType != CORINFO_TYPE_UNDEF)
+#ifdef _TARGET_ARM64_
+        if (corType == CORINFO_TYPE_VALUECLASS)
+        {
+            // This is a vector type.
+            // HVAs are only supported on ARM64, and only for sizes of 8 or 16 bytes.
```

Comment:
Could you clarify on the "only for sizes of 8 or 16 bytes"?

The spec says:
> An Homogeneous Short-Vector Aggregate (HVA) is an Homogeneous Aggregate with a Fundamental Data Type that is a Short-Vector type and at most four uniquely addressable members.

A `short-vector` is then classified as a `64-bit` or `128-bit` vector.

Reply:
I'm not sure what clarification you're looking for. It seems like the comment and the spec are saying the same thing. 
Oh - by "sizes" I meant "vector sizes". Is that what needs clarifying?

Reply:
Right, I think this is ultimately in-line with the spec after going through the rest of the code; but it reads a bit funny to me. Not quite sure how to reword it, however....